### PR TITLE
MULE-12802: JMS Publish does not need to use PooledConnectionProvider

### PR DESCRIPTION
**Pull request by estebanwasinger**

### CODE REVIEW

File: src/main/java/org/mule/extensions/jms/internal/common/JmsCommons.java [hunk 1]
```diff
@@ -127,4 +127,37 @@ public static JmsSession createJmsSession(JmsConnection jmsConnection, InternalA
...
+   *
+   * @param session        Session to close
+   * @param sessionManager Session manager to check the transactional status
+   * @param closeables     All the things that can be closed
+   */
+  public static void releaseResources(JmsSession session, JmsSessionManager sessionManager, AutoCloseable... closeables) {
+    stream(closeables).forEach(JmsCommons::closeQuietly);
+
+    if (!session.getAckId().isPresent()
+        && (!sessionManager.getTransactedSession().isPresent() || sessionManager.getTransactedSession().get() != session)) {
```
Comment (alegmarra) [inline src/main/java/org/mule/extensions/jms/internal/common/JmsCommons.java]: `sessionManager.getTransactedSession().isPresent() || sessionManager.getTransactedSession().get() != session)` => clarifying method like `onCurrentTransaction`

File: src/main/java/org/mule/extensions/jms/internal/connection/JmsTransactionalConnection.java [hunk 2]
```diff
@@ -90,6 +88,7 @@ private void executeTransactionAction(String action, SessionAction transactional
     } finally {
       jmsSessionManager.changeTransactionStatus(NONE);
       jmsSessionManager.unbindSession();
+      jmsSession.close();
```
Comment (marianogonzalez) [inline src/main/java/org/mule/extensions/jms/internal/connection/JmsTransactionalConnection.java]: shouldn't this be on a finally or something? Can any of the methods above fail?

Comment (alegmarra) [inline src/main/java/org/mule/extensions/jms/internal/connection/JmsTransactionalConnection.java]: this is in a finally

Comment (marianogonzalez) [inline src/main/java/org/mule/extensions/jms/internal/connection/JmsTransactionalConnection.java]: i mean a nested finally, to make sure the session is closed even if any of the two statements above fail

**Pull request merged**